core/actor/src/main/scala/net/liftweb/actor/LAFuture.scala

```diff
@@ -295,13 +300,13 @@ object LAFuture {
     }
   }
 
-  private def executeWithObservers(scheduler: LAScheduler, f: () => Unit) {
+  private def executeWithObservers(scheduler: LAScheduler, f: () => Unit, wrapper:CommonLoanWrapper) {
     val cur = threadInfo.get()
     scheduler.execute(() => {
       val old = threadInfo.get()
       threadInfo.set(cur)
       try {
-        f()
+        wrapper(f())
```

Comment:
I can go either way… `Option` seems like the most correct encoding of what we're doing, from a type perspective, I think, which is why I prefer it.